crates/tests/winrt/tests/async.rs

```diff
@@ -1,17 +1,19 @@
+use std::convert::TryFrom;
+
 #[test]
 fn async_get() -> windows::core::Result<()> {
     use windows::Storage::Streams::*;
 
     let stream = &InMemoryRandomAccessStream::new()?;
 
-    let writer = DataWriter::CreateDataWriter(stream)?;
+    let writer = DataWriter::CreateDataWriter(&IOutputStream::try_from(stream)?)?;
```

Comment:
This is an example of a parameter needing to cast from some class type to a non-exclusive interface type that should happen implicitily because we know that `InMemoryRandomAccessStream` implements `IOutputStream`.

Reply:
Also its dependent on the version in which the non-exclusive interface was introduced. In this case, we know that the interface was introduced in v1 of the runtime class therefore a `From` will suffice and a `TryFrom` is not required.

Reply:
If we cannot get rid of all the `from` and `try_from` requirements then perhaps we just drop the `Into<Borrowed<T>>` and just use `Borrowed<T>` everywhere. At least then it will be consistent and allow for `object.into()` to work with type inference.